cli/cmd/uninstall/uninstall.go

```diff
@@ -361,10 +430,105 @@ func (c *Command) uninstallHelmRelease(releaseName, namespace, releaseType strin
 		c.UI.Output("Uninstall result: %s", res.Info, terminal.WithInfoStyle())
 		return nil
 	}
+
 	c.UI.Output(fmt.Sprintf("Successfully uninstalled %s Helm release.", releaseType), terminal.WithSuccessStyle())
 	return nil
 }
 
+// fetchCustomResourceDefinitions fetches all Custom Resource Definitions managed by Consul.
+func (c *Command) fetchCustomResourceDefinitions() (*apiextv1.CustomResourceDefinitionList, error) {
+	return c.apiext.ApiextensionsV1().CustomResourceDefinitions().List(c.Ctx, metav1.ListOptions{
+		LabelSelector: "app=consul",
+	})
+}
+
+// fetchCustomResources gets a list of all custom resources deployed in the
+// cluster that are managed by Consul.
+func (c *Command) fetchCustomResources(crds *apiextv1.CustomResourceDefinitionList) ([]unstructured.Unstructured, error) {
+	crs := make([]unstructured.Unstructured, 0)
+	for _, crd := range crds.Items {
+		for _, version := range crd.Spec.Versions {
+			target := schema.GroupVersionResource{
+				Group:    crd.Spec.Group,
+				Version:  version.Name,
+				Resource: crd.Spec.Names.Plural,
+			}
+
+			crList, err := c.dynamic.Resource(target).List(c.Ctx, metav1.ListOptions{})
+			if err != nil {
+				return nil, err
+			}
+			if crList != nil {
+				crs = append(crs, crList.Items...)
+			}
+		}
+	}
+
+	return crs, nil
+}
+
+// deleteCustomResources takes a list of unstructured custom resources and
+// sends a request to each one to be deleted.
+func (c *Command) deleteCustomResources(crs []unstructured.Unstructured, kindToResource map[string]string) error {
+	for _, cr := range crs {
```

Comment:
I don't think I saw UI logging for deletion. It'd be nice to tell a user that we're deleting the CRs unless I missed these logs somewhere

Reply:
The logging could be easy to miss. It's [on line 373](https://github.com/hashicorp/consul-k8s/blob/1f36db0ed79545d2ea8fcb1dfb972d492d61b8b2/cli/cmd/uninstall/uninstall.go#L373). I used the logger that gets passed to Helm because the "Deleting custom resources managed by Consul" info seems to mesh nicely with the other logs directly from Helm (e.g. "Deleting PVCs for..."). Because I wanted them to look the same and be grouped together, I used that logger so that in the future, if we change how that looks, the "Deleting custom resources managed by Consul" logging changes with it.

Reply:
I meant that we should also log the individual resources that get deleted. I thought we do it for other resources like PVCs etc

Reply:
Yes!

Reply:
Ah perfect!